src/id3/v2/tag.rs

```diff
@@ -285,6 +297,52 @@ impl ID3v2Tag {
 
 		(None, None)
 	}
+
+	fn insert_item(&mut self, item: TagItem) {
+		fn set_number<F: FnMut(u32)>(item: &TagItem, mut setter: F) {
+			let text = item.value().text();
+			let number = text.map(str::parse::<u32>);
+
+			match number {
+				Some(Ok(number)) => setter(number),
+				Some(Err(parse_error)) => {
+					log::warn!(
+						"\"{}\" cannot be parsed as number in {:?}: {parse_error}",
+						text.unwrap(),
+						item.key()
+					)
+				},
+				None => {
+					log::warn!("Value does not have text in {:?}", item.key())
+				},
+			}
+		}
+
+		match item.key() {
+			ItemKey::TrackNumber => set_number(&item, |number| self.set_track(number)),
+			ItemKey::TrackTotal => set_number(&item, |number| self.set_track_total(number)),
+			ItemKey::DiscNumber => set_number(&item, |number| self.set_disk(number)),
+			ItemKey::DiscTotal => set_number(&item, |number| self.set_disk_total(number)),
+			_ => {
+				if let Some(frame) = item.into() {
+					if let Some(replaced) = self.insert(frame) {
+						log::warn!("Replaced frame: {replaced:?}");
+					}
+				}
+			},
+		};
+	}
+
+	fn insert_number_pair(&mut self, id: &'static str, number: Option<u32>, total: Option<u32>) {
+		let content = format_number_pair(number, total);
+
+		match content {
```

Comment:
Using `if let Some() = ` would be more idiomatic. I remember that clippy warns when matching on `Option`.

Reply:
Thank you for comment. I fixed the code.